tools/file_packager.py

```diff
@@ -191,17 +198,13 @@
 if not has_preloaded or jsoutput == None:
   assert not separate_metadata, 'cannot separate-metadata without both --preloaded files and a specified --js-output'
 
+# `"__EMSCRIPTEN_PRIVATE_MODULE_EXPORT_NAME_SUBSTITUTION__"` will be replaced with actual code on later stage of the build, this way Closure Compiler will not mangle it
 ret = '''
-var Module;
-'''
-if not no_closure:
-  ret += '''
-if (typeof Module === 'undefined') Module = eval('(function() { try { return Module || {} } catch(e) { return {} } })()');
-'''
-else:
-  ret += '''
-if (typeof Module === 'undefined') Module = {};
+Module = "__EMSCRIPTEN_PRIVATE_MODULE_EXPORT_NAME_SUBSTITUTION__";
 '''
+if no_closure:
+  replacement = "typeof %(EXPORT_NAME)s !== 'undefined' ? %(EXPORT_NAME)s : {}" % {"EXPORT_NAME": export_name}
+  ret = ret.replace('"__EMSCRIPTEN_PRIVATE_MODULE_EXPORT_NAME_SUBSTITUTION__"', replacement)
 
```

Comment:
actually, re-reading this part, I'm not sure I understand why it's right. Maybe add a comment? It seems like if closure is run, then we are definitely being called from emcc, and so the string will be replaced there. If closure is not run, we may either be called by emcc or not - if it is, the string will be replaced, but if not, what happens here?